linera-core/src/client/mod.rs

```diff
@@ -255,18 +255,33 @@ impl<P, S: Storage + Clone> Client<P, S> {
         pending_block: Option<Block>,
         pending_blobs: BTreeMap<BlobId, Blob>,
     ) -> ChainClient<P, S> {
-        let dashmap::mapref::entry::Entry::Vacant(e) = self.chains.entry(chain_id) else {
-            panic!("Inserting already-existing chain {chain_id}");
+        match self.chains.entry(chain_id) {
+            dashmap::mapref::entry::Entry::Vacant(e) => {
+                e.insert(ChainState::new(
+                    known_key_pairs,
+                    admin_id,
+                    block_hash,
+                    timestamp,
+                    next_block_height,
+                    pending_block,
+                    pending_blobs,
+                ));
+            }
+            dashmap::mapref::entry::Entry::Occupied(e) => {
+                let state = e.get();
+                let owners = known_key_pairs
+                    .into_iter()
+                    .map(|kp| Owner::from(kp.public()))
+                    .collect::<BTreeSet<_>>();
+                assert!(state.known_key_pairs().keys().eq(&owners));
+                assert_eq!(state.admin_id(), admin_id);
```

Comment:
Oh wow that is ugly :-|
Perhaps add a TODO and a task?

Reply:
I created https://github.com/linera-io/linera-protocol/issues/2600 for it, and will add a `TODO`.